### PR TITLE
FINERACT-982 - Completely ditch use of Drizzle JDBC Driver after all

### PR DESCRIPTION
**Pull request by ptuomola**
## Description

Building on (and including) the excellent work of @IOhacker, this completes the removal of Drizzle from all parts of the code. This also stops including any JDBC driver in the built jars / wars and instead instructs the user to include their favourite JDBC driver in the classpath when running the JAR / WAR. 

The built docker image still continues to include the MariaDB driver JAR but instructions are provided to replace it with another if the user so prefers. 

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,17 +21,17 @@ If you are interested in contributing to this project, but perhaps don't quite k
...
 
 You can run the required version of the database server in a container, instead of having to install it, like this:
 
-    docker run --name mysql-5.7 -p 3306:3306 -e MYSQL_ROOT_PASSWORD=mysql -d mysql:5.7
+    docker run --name mariadb-10.6 -p 3306:3306 -e MARIADB_ROOT_PASSWORD=mysql -d mariadb:10.6
 
 and stop and destroy it like this:
 
-    docker rm -f mysql-5.7
+    docker rm -f mysql-10.6
```
Comment (awasum) [inline README.md]: Did you mean mariadb-10.6? Typo here?

Comment (ptuomola) [reply README.md]: Yes of course... let me fix

**Comment by awasum** [comment]
I wonder if something like ./gradlew bootRun will still work? And will it be using the default MariaDB client now?
How does this affect tomcat Deployments and reporting for clients downstream?
And you already made sure the war or jar tasks don't include the MariaDB clients?

This looks like a major change and improvement. We will have to document this also on the wiki.

**Comment by awasum** [comment]
I will have to create time to test this today or tomorrow night... But don't let this stop you. @vidakovic, @IOhacker, @avikganguly01, @fynmanoj @rrpawar96 @francisguchie. This looks like something you will like to test, understand and provide feedback on.

**Comment by ptuomola** [comment]
> I wonder if something like ./gradlew bootRun will still work?  
> And will it be using the default MariaDB client now? 

Yes, bootRun will work as I've added the MariaDB client as a dependency. It will be using the default MariaDB client. 

> How does this affect tomcat Deployments and reporting for clients downstream? 

They should add a JDBC client library into their classpath - either Drizzle, or MariaDB, or MySQL driver. I'd assume that most people would already be using MariaDB / MySQL driver with their Tomcat deployment anyway, as Drizzle is hardly production-grade. In such case there will be no impact. 

> And you already made sure the war or jar tasks don't include the MariaDB clients?

I checked the JAR, but can double-check and also check the WAR later tonight.

**Comment by vidakovic** [comment]
note: working on an easier approach for boot run... no need for mariadb
parts after...

**Comment by vorburger** [comment]
> note: working on an easier approach for boot run... no need for mariadb parts after...

MariaDB4j? 😈  https://github.com/vorburger/MariaDB4j

**Comment by ptuomola** [comment]
@awasum any thoughts - should we merge this or not? It's a big change but if we are serious about getting rid of Drizzle then we should go ahead. Otherwise the pull request will just go stale and we need to start from scratch if we later want to do it.

**Comment by awasum** [comment]
> > I wonder if something like ./gradlew bootRun will still work?
> > And will it be using the default MariaDB client now?
> 
> Yes, bootRun will work as I've added the MariaDB client as a dependency. It will be using the default MariaDB client.
> 
> > How does this affect tomcat Deployments and reporting for clients downstream?
> 
> They should add a JDBC client library into their classpath - either Drizzle, or MariaDB, or MySQL driver. I'd assume that most people would already be using MariaDB / MySQL driver with their Tomcat deployment anyway, as Drizzle is hardly production-grade. In such case there will be no impact.
> 
> > And you already made sure the war or jar tasks don't include the MariaDB clients?
> 
> I checked the JAR, but can double-check and also check the WAR later tonight.

Let document this for every release.

**Review by awasum — APPROVED** [review]
Go ahead and merge this @ptuomola .

**Comment by vorburger** [comment]
The upgrade of Flyway which was "snuck in" 😈 to this PR broke Fineract for (ancient, yes) MySQL v5.7,  see https://issues.apache.org/jira/browse/FINERACT-1457.

**Comment by ptuomola** [comment]
Yes - the old version of Flyway did not work with the new version of MariaDB, hence this PR had to include an upgrade of the Flyway as well. I agree it should have been mentioned in the PR description - sorry about this. 

For clarity's sake, this PR also changed the version of Ubuntu that we are building with from 18.04 to 20.04. Hope that does not break anything for anyone.

**Comment by francisguchie** [comment]
> I will have to create time to test this today or tomorrow night... But don't let this stop you. @vidakovic, @IOhacker, @avikganguly01, @fynmanoj @rrpawar96 @francisguchie. This looks like something you will like to test, understand and provide feedback on.

Yes @awasum  I will create time to test and give give feedback

**Comment by francisguchie** [comment]
> > I wonder if something like ./gradlew bootRun will still work?
> > And will it be using the default MariaDB client now?
> 
> Yes, bootRun will work as I've added the MariaDB client as a dependency. It will be using the default MariaDB client.
> 
> > How does this affect tomcat Deployments and reporting for clients downstream?
> 
> They should add a JDBC client library into their classpath - either Drizzle, or MariaDB, or MySQL driver. I'd assume that most people would already be using MariaDB / MySQL driver with their Tomcat deployment anyway, as Drizzle is hardly production-grade. In such case there will be no impact.
> 
> > And you already made sure the war or jar tasks don't include the MariaDB clients?
> 
> I checked the JAR, but can double-check and also check the WAR later tonight.

yes @ptuomola, its true that Drizzle is not production-grade and i last used it in 2017. However in a number of deployments i have been forced to keep it in the lib folder because sometimes i would get errors so hard to explain from a users point. 
This improvement makes work a lot lighter (being sure if the driver i choose to use)

**Comment by francisguchie** [comment]
> Yes - the old version of Flyway did not work with the new version of MariaDB, hence this PR had to include an upgrade of the Flyway as well. I agree it should have been mentioned in the PR description - sorry about this.
> 
> For clarity's sake, this PR also changed the version of Ubuntu that we are building with from 18.04 to 20.04. Hope that does not break anything for anyone.

@ptuomola  & @vorburger  this is great and interesting information for me - been holding onto ubuntu 18.04 yet 20.4 has been calling all this while. 

All i can say is thanks to this great work done. My biggest struggle has been getting to install MySQL 5.7 on 20.4 because getting the war deployments to work on MySQL **that is above 5.7** was another challenge

With this merge am heading straight for 20.4 with its compatible MySQL version

**Comment by LeonTz1** [comment]
@francisguchie Great work well and thanks for sharing we are looking forward to 20.4.

**Pull request merged**